### ISSUE TITLE
`ExecutorTest#test_class_serial_is_unaffected` testing with Ruby 3.2

### ISSUE THREAD

**Issue by voxik** [rails/rails#46712]
### Steps to reproduce

Running the activesupport 7.0.4 test suite against `ruby 3.2.0dev (2022-12-13 master ece6246057) [x86_64-linux]`, I observe the following error:

~~~
  1) Error:
ExecutorTest#test_class_serial_is_unaffected:
ArgumentError: unknown key: class_serial
    /builddir/build/BUILD/activesupport-7.0.4/usr/share/gems/gems/activesupport-7.0.4/test/executor_test.rb:235:in `stat'
    /builddir/build/BUILD/activesupport-7.0.4/usr/share/gems/gems/activesupport-7.0.4/test/executor_test.rb:235:in `test_class_serial_is_unaffected'
~~~

Looking at recent commits, I have tried to apply the e4140140af5832e045bf8196d1bcbe9e5276a8fb, but it does not seems to help.

### Expected behavior

The test is passing testing with Ruby 3.2

### Actual behavior

The test case fails, even trying with e4140140af5832e045bf8196d1bcbe9e5276a8fb

**Comment by voxik** [comment]
BTW can't see Ruby 3.2 in the CI, am I missing something?

**Comment by yahonda** [comment]
I have reproduced this issue locally and found this behavior change has been introduced via https://github.com/ruby/ruby/commit/13bd617ea6fdf72467c593639cf33312a06c330c


Rails CI has not run against Ruby 3.2 dev recently.
Refer to these pull requests https://github.com/rails/rails/pull/46711 and https://github.com/rails/buildkite-config/pull/24 for detail.

**Comment by yahonda** [comment]
`RubyVM.stat(:class_serial)` has been added via https://github.com/rails/rails/pull/26684 for testings.

**Comment by matthewd** [comment]
Ah cool, I think we can just drop that test on 3.2 -- I believe the upstream change means new Rubies wouldn't suffer even if we used the older `instance_eval`-heavy behaviour. (We should still keep it around for < 3.2 for now, though, because IIRC it was quite a performance hit.)

**Comment by yahonda** [comment]
Closed via #46735 46735

**Comment by voxik** [comment]
Thx for analysis. I would have not figured this is due to change in Ruby. And thx for enabling the CI for Ruby 3.2 :clap:
